src/helpers/AuthContext.tsx

```diff
@@ -44,20 +45,21 @@ function userReducer(state: UserSessionData, action: Action): UserSessionData {
     }
 
     case 'LOGIN':
-      const newState = {
+      const newState: UserSessionData = {
         ...state,
         token: action.payload!.token,
         orgMembership: action.payload!.orgMembership,
         appId: action.payload!.appId,
         firstName: action.payload!.firstName,
         lastName: action.payload!.lastName,
-        userName: action.payload!.username,
+        username: action.payload!.username,
         dataGroups: action.payload!.dataGroups,
         roles: action.payload!.roles,
         id: action.payload!.id,
         synapseUserId: action.payload!.synapseUserId,
         isVerified: action.payload!.isVerified,
         demoExternalId: action.payload?.demoExternalId,
+        lastLoginMethod: action.payload?.lastLoginMethod,
```

Comment:
`lastLoginMethod` will allow us to determine whether to redirect the user to Synapse or logout the user.